projects/packages/forms/src/contact-form/class-util.php

```diff
@@ -276,6 +278,25 @@ public static function jetpack_tracks_record_grunion_pre_message_sent( $post_id
 
 		/** This action is documented in jetpack/modules/widgets/social-media-icons.php */
 		do_action( 'jetpack_bump_stats_extras', 'jetpack_forms_message_sent', $extra );
+
+		$form_type = 'post';
+		if ( isset( $extra_values['widget'] ) ) {
+			$form_type = 'widget';
+		} elseif ( isset( $extra_values['block_template'] ) ) {
+			$form_type = 'block_template';
+		} elseif ( isset( $extra_values['block_template_part'] ) ) {
+			$form_type = 'block_template_part';
```

Comment:
What's the benefit of listing template & template part separately? Would combining them let us follow how many submits come via the block vs widget?

Reply:
since we have that date it would be more granular? happy to combine

Reply:
What does the difference mean in practise? If the form was on page or on sidebar/header/footer?

Reply:
oh I see you [elaborated above](https://github.com/Automattic/jetpack/pull/41307#discussion_r1932110092)